src/main/java/com/conveyal/gtfs/validator/OverlappingTripValidator.java

```diff
@@ -33,94 +35,66 @@ public OverlappingTripValidator(Feed feed, SQLErrorStorage errorStorage) {
     @Override
     public void validateTrip(Trip trip, Route route, List<StopTime> stopTimes, List<Stop> stops) {
         if (trip.block_id != null) {
+            // If the trip has a block_id, add a new block interval to the map.
             BlockInterval blockInterval = new BlockInterval();
             blockInterval.trip = trip;
             StopTime firstStopTime = stopTimes.get(0);
             blockInterval.startTime = firstStopTime.departure_time;
             blockInterval.firstStop = firstStopTime;
             blockInterval.lastStop = stopTimes.get(stopTimes.size() - 1);
-            List<BlockInterval> intervals = blockIntervals.get(trip.block_id);
-            if (intervals == null) {
-                intervals = new ArrayList<>();
-                blockIntervals.put(trip.block_id, intervals);
-            }
-            intervals.add(blockInterval);
+            // Construct new list of intervals if none exists for encountered block_id.
+            blockIntervals
+                .computeIfAbsent(trip.block_id, k -> new ArrayList<>())
+                .add(blockInterval);
         }
     }
 
     @Override
     public void complete (ValidationResult validationResult) {
+        // Iterate over each block and determine if there are any trips that overlap one another.
         for (String blockId : blockIntervals.keySet()) {
             List<BlockInterval> intervals = blockIntervals.get(blockId);
-            Collections.sort(intervals, Comparator.comparingInt(i -> i.startTime));
-            int i2offset = 0;
+            intervals.sort(Comparator.comparingInt(i -> i.startTime));
+            // Iterate over each interval (except for the last) comparing it to every other interval (so the last interval
+            // is handled through the course of iteration).
             // FIXME this has complexity of n^2, there has to be a better way.
```

Comment:
I think this FIXME comment could be removed. Also, this probably isn't a bottleneck since there are probably not millions of different intervals per block.